server/src/main/java/org/opensearch/index/codec/CodecServiceConfig.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.mapper.MapperService;
+
+import java.util.Objects;
+
+/**
+ * The configuration parameters necessary for the {@link CodecService} instance construction.
+ */
+public final class CodecServiceConfig {
+    private final IndexSettings indexSettings;
```

Comment:
We could, I believe the presence of the index settings could be beneficial, otherwise would leave that implementors to pass the index settings from `EnginePlugin` to `CodecService` instance if needed